tests/testlogin.js

```diff
@@ -0,0 +1,79 @@
+//const { assert } = require("console");
+import LoginPage from '../pages/login.page';
+import HomePage from '../pages/home.page';
+
+describe ('TestLogin', async () => {
+
+    it('Should login successfully', async () => {
+
+        await LoginPage.open('login');
+        await LoginPage.loginAs('jsmith@demo.io', 'Demo123!');
+        await expect(HomePage.welcomeUser).toHaveTextContaining('Welcome Josh');
+        
+    });
+
+    it('Should login successfully with Remember me chechbox selected', async () => {
+
+        await LoginPage.open('login');
+        await LoginPage.selectRememberMe('jsmith@demo.io', 'Demo123!');
+        await expect(HomePage.welcomeUser).toHaveTextContaining('Welcome Josh');
+        
+    });
+
+    it ('Should not login with invalid username and correct password', async () => {
+
+        await LoginPage.open('login');
+        await LoginPage.loginAs('andrea@demo.io', 'Demo123!');
+        expect(await $('badge badge-pill badge-danger')).toHaveTextContaining('Error');
```

Comment:
By using the page object pattern, it implies the tests won't interact with raw elements directly. You are mixing levels of abstraction here: Page object methods with raw element selectors. Represent this error message element as an attribute from the LoginPage class, like you did with `usernameInput`, and use it for this assertion, so you'll end up with something like:

```javascript
expect(await LoginPage.statusMessage).toHaveTextContaining('Error');
```